android/src/main/java/com/applovin/reactnative/AppLovinMAXAdView.java

```diff
@@ -155,66 +164,86 @@ public void attachAdView()
             return;
         }
 
-        // Run in next run loop when view is laid out and `getWidth()` / `getHeight()` has appropriate values
-        post( new Runnable()
+        if ( TextUtils.isEmpty( adUnitId ) )
+        {
+            AppLovinMAXModule.e( "Attempting to attach MAAdView without Ad Unit ID" );
+            return;
+        }
+
+        if ( adFormat == null )
+        {
+            AppLovinMAXModule.e( "Attempting to attach MAAdView without ad format" );
+            return;
+        }
+
+        // Re-assign in case of race condition
+        final String adUnitId = this.adUnitId;
+        final MaxAdFormat adFormat = this.adFormat;
+
+        // Run after 0.25 sec delay to allow all properties to set
+        postDelayed( new Runnable()
         {
             @Override
             public void run()
             {
-                if ( adView != null ) return;
+                if ( adView != null )
+                {
+                    AppLovinMAXModule.e( "Attempting to re-attach with existing MAAdView: " + adView );
+                    return;
+                }
 
-                // If ad unit id and format has been set - create and attach AdView
-                if ( !TextUtils.isEmpty( adUnitId ) && adFormat != null )
+                AppLovinMAXModule.d( "Attaching MAAdView..." );
+
+                adView = new MaxAdView( adUnitId, adFormat, AppLovinMAXModule.getInstance().getSdk(), currentActivity );
+                adView.setListener( AppLovinMAXModule.getInstance() );
+                adView.setRevenueListener( AppLovinMAXModule.getInstance() );
```

Comment:
How come

```

                if ( placement != null )
                {
                    adView.setPlacement( placement );
                }
                if ( customData != null )
                {
                    adView.setCustomData( customData );
                }
```

changes were not ported over?

Reply:
This is Boolean not boolean.    It checks whether these values are actually set or not.   Should we do the same on the iOS side?

Reply:
This actually brought up the default values for the props.   In the current implementation, no default value is defined, so the properties are set only when they have values.   The new implementation's behavior is the same but not for boolean values in iOS.   So defined default values in Javascript.